File: src/main/java/com/example/serveroptvisualizer/concert/Concert.java
package com.example.serveroptvisualizer.concert;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name = "concert")
public class Concert {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String title;

  @Column(nullable = false)
  private String venue;

  @Column(columnDefinition = "text")
  private String description;

  @Column(nullable = false)
  private LocalDateTime createdAt;

  protected Concert() {}

  public Concert(String title, String venue, String description, LocalDateTime createdAt) {
    this.title = title;
    this.venue = venue;
    this.description = description;
    this.createdAt = createdAt;
  }

  public Long getId() {
    return id;
  }

  public String getTitle() {
    return title;
  }

  public String getVenue() {
    return venue;
  }

  public String getDescription() {
    return description;
  }

  public LocalDateTime getCreatedAt() {
    return createdAt;
  }
}

File: src/main/kotlin/com/example/serveroptvisualizer/concert/Concert.kt
package com.example.serveroptvisualizer.concert

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "concert")
class Concert(
  @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Long? = null,
  @Column(nullable = false) var title: String = "",
  @Column(nullable = false) var venue: String = "",
  @Column(columnDefinition = "text") var description: String? = null,
  @Column(nullable = false) var createdAt: LocalDateTime = LocalDateTime.now(),
)
